mirrord/layer/tests/apps/dns_resolve/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use std::{ffi::CString, ptr, str};
+
+fn get_ip_addresses(node: &str) -> (Vec<String>, i32) {
+    let node = CString::new(node).unwrap();
+    let service = ptr::null_mut::<libc::c_char>();
+
+    let hints: libc::addrinfo = unsafe { std::mem::zeroed() };
+
+    let mut res: *mut libc::addrinfo = ptr::null_mut();
```

Comment:
btw you can probably lookup use `.to_socket_addrs` instead of doing libc stuff (I checked, they use `getaddrinfo` underneath..)

Reply:
oh ): yes I saw on stackoverflow when i was doing recv_from, i should use that

Reply:
i keep getting `thread 'main' panicked at 'Error: invalid socket address', mirrord/layer/tests/apps/dns_resolve/src/main.rs:7:19
stack backtrace:`

rust app:

```rs
use std::net::{ToSocketAddrs, SocketAddr};

fn get_ip_addresses(node: &str) -> (Vec<String>, i32) {
    print!("Looking up IP addresses for {}... ", node   );
    let addresses: Vec<SocketAddr> = match node.to_socket_addrs() {
        Ok(addrs) => addrs.collect(),
        Err(e) => panic!("Error: {}", e),
    };

    let addresses_str = addresses
        .iter()
        .map(|addr| addr.ip().to_string())
        .collect::<Vec<String>>();

    (addresses_str, 0)
}

fn test_getaddrinfo() {
    let domain_x = "example.com";
    let domain_y = "nonexistent-domain";

    let (addresses_x, result_x) = get_ip_addresses(domain_x);
    assert!(!addresses_x.is_empty(), "Expected addresses for domain X");
    assert_eq!(result_x, 0, "Expected success for domain X");

    let (addresses_y, result_y) = get_ip_addresses(domain_y);
    assert!(addresses_y.is_empty(), "Expected no addresses for domain Y");
    assert_eq!(result_y, libc::EAI_FAIL, "Expected failure for domain Y");
}

fn main() {
    test_getaddrinfo();
}
```

the discussion i mentioned:
https://stackoverflow.com/questions/68042772/how-to-do-a-dns-lookup-without-any-crates

> That will just resolve an IP address. But DNS is so much more powerful - there are TXT,MX,PTR, CNAME,etc, all kinds of different record types which you cannot access with this approach

i am not 100% sure about how DNS queries work at a lower level, but i am assuming its not enough and we still probably need getaddrinfo?

Reply:
We don't care about it. Also, getaddrinfo is A records afaik..